File: 20180709/TransposeMatrix.java
class Solution {
    public int[][] transpose(int[][] A) {
        int M = A.length;
        if (M < 1){
            int[][] res = {{}};
            return res;
        }
        int[][] result = new int[A[0].length][];
        int N = A[0].length;
        List<Integer> expand = new ArrayList<>();
        for(int row = 0; row < M; row ++){
            for(int col=0; col < N; col++){
                expand.add(A[row][col]);
            }
        }
        for(int i = 0; i < N; i ++){
            int[] match = new int[A.length];
            for (int j = 0; j < M; j++){
                match[j] = expand.get(i +(j * N));
            }
            result[i] = match;
        }
        return result;
    }
}

File: 20180709/TransposeMatrix.py
class Solution(object):
    def transpose(self, A):
        """
        :type A: List[List[int]]
        :rtype: List[List[int]]
        """
        if len(A) < 1:
            return []
        # for col in range(len(A[0])):
        #     tmp = []
        #     for raw in range(len(A)):
        #         tmp.append(A[raw][col])
        #     res.append(tmp)
        # return res
        expand = [item for row in A for item in row]
        col = len(A[0])
        row = len(A)
        return [[expand[i + (r *col)] for r in range(row)] for i in range(col)]
